src/trace/Kconfig

```diff
@@ -31,4 +31,59 @@ config TRACEM
 	help
 	  Sending all traces by mailbox additionally.
 
+config TRACE_FILTERING
+	bool "Trace filtering"
+	depends on TRACE
+	default y
+	help
+		Filtering of trace messages based on their verbosity level and velocity.
+
+config TRACE_FILTERING_VERBOSITY
+	bool "Filter by verbosity"
+	depends on TRACE_FILTERING
+	default y
+	help
+		Filtering by log verbosity level, where maximum verbosity allowed is specified for each
+		context and may be adjusted in runtime.
+
+config TRACE_FILTERING_ADAPTIVE
+	bool "Adaptive rate limiting"
+	depends on TRACE_FILTERING
+	default y
+	help
+		Adaptive filtering of trace messages, tracking up to CONFIG_TRACE_RECENT_ENTRIES_COUNT,
+		suppressing all repeated messages for up to CONFIG_TRACE_RECENT_TIME_THRESHOLD cycles.
+
+config TRACE_RECENT_ENTRIES_COUNT
+	int "Amount of entries considered recent"
+	depends on TRACE_FILTERING_ADAPTIVE
+	default 5
+	help
+		Recent entries are used to determine whether currently processed message was sent in the
+		past. Size of that filtering window affects effectiveness and performance. More recent
+		entries allow to better filter repetetive messeges out, but also slightly decrease
+		performance due to increased number of comparisions necessary.
+
+config TRACE_RECENT_TIME_THRESHOLD
+	int "Period of time considered recent (microseconds)"
+	depends on TRACE_FILTERING_ADAPTIVE
+	default 1500
+	help
+		Period of time during which entries are tracked and will be suppressed if reported again.
+
+config TRACE_RECENT_PROLONG
+	bool "Prolong suppression if retriggered"
+	depends on TRACE_FILTERING_ADAPTIVE
+	default y
+	help
+		With this option enabled, suppression timeout is reset each time repeated message is sent,
+		up to maximum suppresion time CONFIG_TRACE_RECENT_MAX_TIME.
+
+config TRACE_RECENT_MAX_TIME
+	int "Maximum period of time that message can be suppressed (microseconds)"
+	depends on TRACE_RECENT_PROLONG
+	default 5000000
```

Comment:
does this really have to be configurable? Maybe better not to overwhelm the user with too many configuration options? :-)

Reply:
Well, this config option allows you to set "how fast will the trace buffer be filled with 1ms spam."
Assuming, that buffer is 10kB (don't remember how big it is), log entry is max 36B, we have enough space for at least 284 log entries. With current default value (5 seconds), the buffer could be full in a little over 10 minutes, with just 1 spamming trace message, because every 5s we would inform about suppression and allow 1 unsuppressed message through (again, this is assuming that kernel keeps 10kB buffer for log entries).
I don't know if these values are good enough to protect us from worthless logs in bug reports. Also I don't want to keep pseudo-magic defines in the code, when we can easily extract them and allow control over them.

Reply:
KConfig is good for me as they are more obvious as places to tune the output for a given debug issue.